test/onnx/test_utility_funs.py

```diff
@@ -675,6 +675,29 @@ def forward(self, x):
 
         np.testing.assert_allclose(ratio_pytorch, ratio_ort, rtol=0.01, atol=0.01)
 
+    def test_flatten_dynamic_axes(self):
+        class MyModule(torch.nn.Module):
+            def forward(self, x):
+                return torch.flatten(x, start_dim=2, end_dim=3)
+
+        batch_size = 3
+        x = torch.randn(batch_size, 5, 4, 5)
+        model = MyModule()
+        f = io.BytesIO()
+        torch.onnx.export(model, (x,), f, input_names=['input'], output_names=['output'],
+                          dynamic_axes={'input' : {0 : 'batch_size'},
+                                        'output' : {0 : 'batch_size'}})
+
+        x_ort = torch.randn(5, 5, 4, 5)
+        ort_sess = onnxruntime.InferenceSession(f.getvalue())
+        ort_inputs = {ort_sess.get_inputs()[0].name : x_ort.cpu().numpy()}
+        ort_outs = ort_sess.run(None, ort_inputs)[0]
+
+        y = model(x_ort)
+        pytorch_outs = y.cpu().numpy()
+
+        [np.testing.assert_allclose(out, ort_out, rtol=1e-3, atol=1e-7) for out, ort_out in zip(pytorch_outs, ort_outs)]
```

Comment:
Should we put this test in `test_pytorch_onnx_onnxruntime.py`? It looks it's just doing the usual stuff: export and run model in ort and compare results with pytorch.

Reply:
the reason why I have put the test in test_utility_funs is to check if model exported with input of shape (3, 5, 4, 5) could run in ort with input of shape (5, 5, 4, 5). I don't think I can do that in test_pytorch_onnx_onnxruntime.py.

Reply:
actually you can :)
please refer to this example in `test_pytorch_onnx_onnxruntime.py` https://github.com/pytorch/pytorch/blob/6debc28964872e26724764e213c40db366327f95/test/onnx/test_pytorch_onnx_onnxruntime.py#L445